erizo/src/erizo/rtp/RRGenerationHandler.cpp

```diff
@@ -69,174 +61,115 @@ int RRGenerationHandler::getVideoClockRate(uint8_t payload_type) {
+        "dlsr: %u, lsr: %u", connection_->toLog(), selected_packet_info->ssrc, selected_packet_info->type,
+        rtcp_head.getLostPackets(), rtcp_head.getFractionLost(), rtcp_head.getSeqnumCycles(),
+        rtcp_head.getHighestSeqnum(), rtcp_head.getJitter(), rtcp_head.getDelaySinceLastSr(),
+        rtcp_head.getLastSr());
+    uint16_t selected_interval = selected_packet_info->type == VIDEO_PACKET ? RTCP_VIDEO_INTERVAL : RTCP_AUDIO_INTERVAL;
+    selected_packet_info->next_packet_ms = now + getRandomValue(0.5 * selected_interval, 1.5 * selected_interval);
   }
 }
 
+
 void RRGenerationHandler::handleSR(std::shared_ptr<dataPacket> packet) {
   RtcpHeader *chead = reinterpret_cast<RtcpHeader*>(packet->data);
-
-    if (chead->getSSRC() == video_rr_.ssrc) {
-      video_rr_.last_sr_mid_ntp = chead->get32MiddleNtp();
-      video_rr_.last_sr_ts = packet->received_time_ms;
-      uint32_t expected = video_rr_.extended_seq- video_rr_.base_seq + 1;
-      video_rr_.lost = expected - video_rr_.p_received;
-      uint8_t fraction = 0;
-      uint32_t expected_interval = expected - video_rr_.expected_prior;
-      video_rr_.expected_prior = expected;
-      uint32_t received_interval = video_rr_.p_received - video_rr_.received_prior;
-      video_rr_.received_prior = video_rr_.p_received;
-      uint32_t lost_interval = expected_interval - received_interval;
-      if (expected_interval != 0 && lost_interval > 0) {
-        fraction = (lost_interval << 8) / expected_interval;
-      }
-      video_rr_.frac_lost = fraction;
-      sendVideoRR();
-    }
-
-    if (chead->getSSRC() == audio_rr_.ssrc) {
-      audio_rr_.last_sr_mid_ntp = chead->get32MiddleNtp();
-      audio_rr_.last_sr_ts = packet->received_time_ms;
-      uint32_t expected = audio_rr_.extended_seq- audio_rr_.base_seq + 1;
-      audio_rr_.lost = expected - audio_rr_.p_received;
-      uint32_t fraction = 0;
-      uint32_t expected_interval = expected - audio_rr_.expected_prior;
-      audio_rr_.expected_prior = expected;
-      uint32_t received_interval = audio_rr_.p_received - audio_rr_.received_prior;
-      audio_rr_.received_prior = audio_rr_.p_received;
-      uint32_t lost_interval = expected_interval - received_interval;
-      if (expected_interval != 0 && lost_interval > 0) {
-        fraction = (lost_interval << 8) / expected_interval;
-      }
-      audio_rr_.frac_lost = fraction;
-      sendAudioRR();
-    }
+  auto rr_packet_pair = rr_info_map_.find(chead->getSSRC());
+  if (rr_packet_pair == rr_info_map_.end()) {
+    ELOG_DEBUG("%s message: handleRtpPacket ssrc not found, ssrc: %u", connection_->toLog(), chead->getSSRC());
+    return;
+  }
+  std::shared_ptr<RRPackets> selected_packet_info = rr_packet_pair->second;
+  selected_packet_info->last_sr_mid_ntp = chead->get32MiddleNtp();
+  selected_packet_info->last_sr_ts = packet->received_time_ms;
+  uint32_t expected = selected_packet_info->extended_seq- selected_packet_info->base_seq + 1;
```

Comment:
also separate expected, expected_interval, received_interval, lost_interval and fraction sections with blank lines.